packages/dd-trace/src/exporters/agent/writer.js

```diff
@@ -1,129 +1,74 @@
 'use strict'
 
 const platform = require('../../platform')
-const log = require('../../log')
-const encode = require('../../encode')
 const tracerVersion = require('../../../lib/version')
+const msgpack = require('msgpack-lite')
 
-const MAX_SIZE = 8 * 1024 * 1024 // 8MB
-const METRIC_PREFIX = 'datadog.tracer.node.exporter.agent'
+const Writer05 = require('./writer-0.5')
+const Writer04 = require('./writer-0.4')
+const BaseWriter = require('./base-writer')
```

Comment:
Having all these writers + the current file which is also a writer is confusing.

Reply:
What would you prefer?

Reply:
Maybe I could call this one `WriterProxy`? Kind of like our span proxy.